lib/sensors/accelerometer.cpp

```diff
@@ -0,0 +1,117 @@
+#include "accelerometer.hpp"
+
+namespace hyped::sensors {
+
+Accelerometer::Accelerometer(std::uint8_t mux_address_ess, io::I2c &i2c, core::ILogger &log)
+    : address_(mux_address_ess),
+      log_(log),
+      i2c_(i2c)
+{
+}
+
+Accelerometer::~Accelerometer()
+{
+}
+
+std::optional<std::int16_t> Accelerometer::getRawAccelerationX()
+{
+  std::optional<std::uint8_t> resultLow  = i2c_.readByte(address_, 0x28);
+  std::optional<std::uint8_t> resultHigh = i2c_.readByte(address_, 0x29);
+
+  if (!resultLow.has_value() || !resultLow.has_value()) { return std::nullopt; }
+
+  std::optional<std::int16_t> rawX = (resultHigh.value() << 8) | resultLow.value();
+
+  return rawX;
+}
+
+std::optional<std::int16_t> Accelerometer::getRawAccelerationY()
+{
+  std::optional<std::uint8_t> resultLow  = i2c_.readByte(address_, 0x2A);
+  std::optional<std::uint8_t> resultHigh = i2c_.readByte(address_, 0x2B);
+
+  if (!resultLow.has_value() || !resultLow.has_value()) { return std::nullopt; }
+
+  std::optional<std::int16_t> rawX = (resultHigh.value() << 8) | resultLow.value();
+
+  return rawX;
+}
+
+std::optional<std::int16_t> Accelerometer::getRawAccelerationZ()
+{
+  std::optional<std::uint8_t> resultLow  = i2c_.readByte(address_, 0x2C);
+  std::optional<std::uint8_t> resultHigh = i2c_.readByte(address_, 0x2D);
+
+  if (!resultLow.has_value() || !resultLow.has_value()) { return std::nullopt; }
+
+  std::optional<std::int16_t> rawX = (resultHigh.value() << 8) | resultLow.value();
+
+  return rawX;
+}
+
+std::optional<core::acceleration_struct> Accelerometer::read()
+{
+  /* check to see if the values are ready to be read */
+  std::optional<std::uint8_t> result = i2c_.readByte(address_, 0x27);
+
+  if (!result.has_value()) { return std::nullopt; }
+
+  // check if sensor says that data is ready to be read
+  if (result.value() == 0x00) {
+    log_.log(core::LogLevel::kFatal, "acceleration data not ready yet to be read");
+    return std::nullopt;
+  }
+
+  // x axis
+  std::optional<std::int16_t> resultX = getRawAccelerationX();
```

Comment:
`const` here and for every other result too, also `result_X` but open to better naming